router/worker.go

```diff
@@ -308,13 +310,20 @@ func (w *worker) transform(routerJobs []types.RouterJobT) []types.DestinationJob
 			w.rt.logger.Debugn("traceParent is empty during router transform", logger.NewIntField("jobId", job.JobMetadata.JobID))
 		}
 	}
-	w.rt.routerTransformInputCountStat.Count(len(routerJobs))
-	destinationJobs := w.rt.transformer.Transform(
-		transformer.ROUTER_TRANSFORM,
-		&types.TransformMessageT{Data: routerJobs, DestType: strings.ToLower(w.rt.destType)},
-	)
-	w.rt.routerTransformOutputCountStat.Count(len(destinationJobs))
-	w.recordStatsForFailedTransforms("routerTransform", destinationJobs)
+	destinationJobs := make([]types.DestinationJobT, 0)
+	destinationIDRouterJobsMap := lo.GroupBy(routerJobs, func(job types.RouterJobT) string {
+		return job.Destination.ID
+	})
+	for _, destinationIDRouterJobs := range destinationIDRouterJobsMap {
```

Comment:
Ignoring oauth for a moment, can transformer itself handle transformation of jobs for different destination ids?

Reply:
transformer can handle different destination ids events, the problem is the oauth v2 interceptor is written in such a way that it assumes that all events belong to same destination id

Reply:
this increases the number of calls to transformer but it doesn't effect the out going requests as we anyway group by destination in transformer more over this only effect free tier as most of the other tiers will run destination isolation.

Reply:
We are also planning to refactor the OAuth v2, but that is a slightly bigger change and requires the transformer service change as well, so we will take it up once this is done.

Reply:
Doing 1 http call with n events in the payload is always going to be better than doing n http calls with 1 event in the payload, that's why we should avoid it if possible.